### ISSUE TITLE
Pip version in the latest release too low

### ISSUE THREAD

**Issue by BlaBlaBlaWang** [mldbai/mldb#954]
Hello, dear developers. I have tried to build MLDB according to Building.md inside the master branch, using the toolchain of gcc. However, when I began to run commands of  "make", many errors occurred.

I firstly tried to fix the errors by modifying the files according to the error logs, however, I was finally stuck at the cffi installation error:
```
Running setup.py install for cffi ... error
...
c/call_python.c:6:56: error: dereferencing pointer to incomplete type ‘PyInterpreterState’ {aka ‘struct _is’} 
6 |   PyObject *m, *modules = PyThreadState_GET()->interp->modules;"
```

I thought I shoud blame the cPython or other modules, and reinstall them by myself, but the Makefile would reinstall all needed modules, referring from the build logs.

Then the notification down the error indicate reminded me:
```
You are using pip version 8.0.2, however version 21.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.  
```

Since the make process would set up a virtual environment and overwrite my `21.3.1` installation inside contemporary virtual environment, I find the Makefile sentences for the assigned pip version `8.0.2` and change them to `21.3.1`, which is the latest version of pip till this Issue. Then though the environment was rebuilt, all prevoius errors disappeared!!! Though I am still pacing the following building steps, I think this problem could be perfectly fixed by simply upgrading the pip version.

Back to the master branch, the commits are mainly in-pace with the pip upgrading,  but the release_latest branch recommended by the Building.md has not received the update. Could you please upgrade the pip version inside the build files of release_latest branch? Or could you please find other solutions? That would help a great deal to freshmen like me, thanks!

**Comment by jeremybarnes** [comment]
Hi there!  Sorry for the slow response.  As you guessed, the `release_latest` branch is not at all up to date... please use the `master` branch instead.  I'll update `Building.md`; thanks for the heads-up!

**Comment by jeremybarnes** [comment]
(I will update the build instructions over the next few days).

**Comment by shijianjian** [comment]
@jeremybarnes  Is there any reason that there is no new releasing since 2017?

**Comment by jeremybarnes** [comment]
@shijianjian Yes, because there is no longer anybody working on it as their job (the company that developed MLDB was sold in 2017, and the new owners decided not to continue development).
